### PR TITLE
Afd 110 edit book logic

### PR DESCRIPTION
**Pull request by TheGuardingDark**
Edit Book page allows you to edit the book details. You can reset the form to the original details. Submitting will take you to the bookshelf.

### CODE REVIEW

**Review by aperezsantos — DISMISSED** [review]
So far, everything functions as expected, except for the mentioned edit image feature (line 26). Glad you made a note of it in your code. It helped me recognize that as something you're still working on.

I would also take a look at your footer. Moving b/w desktop and mobile view, shortens its width for me. I believe it has to do with the EditBook.grid component.

Also between about 480 and and 765 (width), the bottom two buttons disappear behind the footer.

File: src/components/BookForm.jsx [hunk 1]
```diff
@@ -8,39 +9,60 @@ import EmptyCard from './EmptyCard';
...
+  const path = pathname.pathname;
+  const editPath = `/editBook/${id}`;
+
   const [reset, setReset] = useState(false);
   const [validTitle, setValidTitle] = useState(true);
   const [validAuthor, setValidAuthor] = useState(true);
-  const titleRef = useRef();
-  const authorRef = useRef();
+
+  const titleRef = document.getElementById('title');
```
Comment (scullenDev) [inline src/components/BookForm.jsx]: Curious if you had issues w/the original implementation of useRef here?

File: src/components/BookForm.jsx [hunk 2]
```diff
@@ -8,39 +9,60 @@ import EmptyCard from './EmptyCard';
...
+  useEffect(() => {
+    if (existingBook) {
+      setFormObj(existingBook);
+    }
+  }, [existingBook]);
+
+  const { rating, title, author, synopsis, pages, published } = formObj;
+
+  const resetForm = () => {
+    if (id) {
```
Comment (scullenDev) [inline src/components/BookForm.jsx:44]: I love that resetting repopulates the book info in Edit View!

File: src/components/BookForm.jsx [hunk 3]
```diff
@@ -8,39 +9,60 @@ import EmptyCard from './EmptyCard';
...
   const [reset, setReset] = useState(false);
   const [validTitle, setValidTitle] = useState(true);
   const [validAuthor, setValidAuthor] = useState(true);
-  const titleRef = useRef();
-  const authorRef = useRef();
+
+  const titleRef = document.getElementById('title');
+  const authorRef = document.getElementById('author');
 
   const [formObj, setFormObj] = useState({
```
Comment (scullenDev) [inline src/components/BookForm.jsx]: I'm recommend storing the formObj defaults in a variable (preferably within this file but ABOVE the component) so you don't need to duplicate them here and in the reset.

File: src/components/Stars.jsx [hunk 4]
```diff
@@ -10,18 +11,31 @@ const Stars = ({
...
+      setReset(false);
+    } else {
+      setCurrentRating(0);
+      setReset(false);
+    }
   }, [reset]);
 
+  useEffect(() => {
+    if (id) {
+      getBook(id)
```
Comment (scullenDev) [inline src/components/Stars.jsx]: Would it be feasible to pass this book.rating value in from the parent component(s)? It's not ideal that you're making HTTP calls from the Stars component and whatever parent component it resides in (Details, Edit, etc). Please see if a refactor is possible to refactor.

**Review by scullenDev — CHANGES_REQUESTED** [review]
A few recommendations/comments... great work though, Liv.

File: src/components/BookForm.jsx (hunk 1 above)
Comment (TheGuardingDark) [reply src/components/BookForm.jsx]: I actually forgot to change it back. I have now and it seems to be working.

File: src/components/Stars.jsx (hunk 4 above)
Comment (TheGuardingDark) [reply src/components/Stars.jsx]: You got it

**Review by scullenDev — APPROVED** [review]
Nice work, Liv!

**Pull request merged**